### PR TITLE
Update dart builds to include packages and drop dart2dart

### PR DESCRIPTION
**Pull request by butlermatt**

### CODE REVIEW

File: tool/build.dart [hunk 1]
```diff
@@ -262,52 +268,32 @@ _main(List<String> argv) async {
...
-        "build/bin/${mainFile.path
-          .split('/')
-          .last}",
-        "--output-type=dart",
-        "--categories=Server"
-      ], writeToBuffer: true);
-      if (cmfr.exitCode != 0) {
-        await fail("DSLink ${name}: Failed to build main file.\n${cmfr.output}");
-      }
+      if (pkgs != null && pkgs.length > 0) {
```
Comment (Pacane) [inline tool/build.dart]: isNotEmpty

File: tool/util/path.dart [hunk 2]
```diff
@@ -0,0 +1,78 @@
+
```
Comment (Pacane) [inline tool/util/path.dart]: dartfmt

File: tool/util/path.dart [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+String fromUri(uri) {
+  if (uri is String) uri = Uri.parse(uri);
+  return Uri.decodeComponent((uri as Uri).path);
+}
+
+String dirname(String path) {
+  var pt = path.trim();
```
Comment (Pacane) [inline tool/util/path.dart:7]: doesn't this exist with the path package?

**Review by Pacane — APPROVED** [review]

File: tool/util/path.dart (hunk 2 above)
Comment (butlermatt) [reply tool/util/path.dart]: Done

File: tool/util/path.dart (hunk 3 above)
Comment (butlermatt) [reply tool/util/path.dart:7]: Yes, but this script isn't run with pub at all, just executed like a shell script so the entire build would need to be reorganized to make pub packages work.

Initially it was in the path package in my first version of the script on just building one package. It wasn't until I tried to add it to this script that I realized pub support wasn't included. That's why I chose to implement this sub-section of path package the same way it's implemented in the package

**Pull request merged**